File: node/fizzbuzz.js

function isFizz(num) {
    if (num % 3 == 0) {
        return true
    }
    else {
        return false
    }
};


function isBuzz(num) {
    if (num % 5 == 0) {
        return true
    }
    else {
        return false
    }
};

function fizzbuzz(num) {
    var fizz = isFizz(num)
    var buzz = isBuzz(num)
    if (fizz && buzz) {
        return 'FizzBuzz'
    }
    if (fizz) {
        return 'Fizz'
    }
    if (buzz) {
        return 'Buzz'
    }
    else {
        return num.toString()
    }
};

function runfizzbuzz() {
    for (var i = 1; i < 100; i++) {
        console.log(fizzbuzz(i))
    }
};

exports.isFizz = isFizz
exports.isBuzz = isBuzz
exports.fizzbuzz = fizzbuzz

File: python/naive-fizzbuzz/fizzbuzz.py


def isFizz(num):
    return num % 3 ==  0

def isBuzz(num):
    return num % 5 == 0

def fizzbuzz(num):
    fizz = isFizz(num)
    buzz = isBuzz(num)
    if fizz & buzz:
        return 'FizzBuzz'
    elif fizz:
        return 'Fizz'
    elif buzz:
        return 'Buzz'
    else:
        return num

if __name__ == '__main__':
    for x in range(100):
        print(fizzbuzz(x))